compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1109,8 +1109,13 @@ object desugar {
           case Block(Nil, expr) => expr // important for interpolated string as patterns, see i1773.scala
           case t => t
         }
-
-        Apply(Select(Apply(Ident(nme.StringContext), strs), id), elems)
+        // This is a deliberate departure from scalac, where StringContext is not rooted (See #4732)
+        val rootedStringContext = {
+          val rootPkg = Ident(nme.ROOTPKG)
+          val scalaPkg = Select(rootPkg, nme.scala_)
+          Select(scalaPkg, nme.StringContext)
+        }
+        Apply(Select(Apply(rootedStringContext, strs), id), elems)
```

Comment:
Maybe `rootedStringContext` can be replaced with `ref(defn.StringContextModule)`

Reply:
No, Desugar produces untyped trees

Reply:
Oh, I missed that. Indore my comment